### PR TITLE
Ruby 3 Upgrade

### PR DESCRIPTION
**Pull request by iangmaia**
## What does it do?

Building on top of https://github.com/wordpress-mobile/release-toolkit/pull/492 and related PRs, this PR finally upgrades `release-toolkit`'s Ruby version to `3.2.2`, also fixing all reported Rubocop offences (see commits breakdown) and updates known issues (`.rubocop_todo.yml`).

**Note**: This PR should be merged only when we're ready to support Ruby 3 on all client projects (or at least when we're sure they will jump straight to the `release-toolkit` Ruby 3 version).

## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the appropriate existing `###` subsection of the existing `## Trunk` section.
- [x] If applicable, add an entry in [the `MIGRATION.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/MIGRATION.md) to describe how the changes will affect the migration from the previous major version and what the clients will need to change and consider.

### CODE REVIEW

**Comment by crazytonyli** [comment]
All of our apps and most projects that are in active development have been migrated to Ruby 3.2.2. I think it's time for release-toolkit to make the transition too?

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please add an entry in the CHANGELOG.md file to describe the changes made by this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by iangmaia** [comment]
Great @crazytonyli, thanks for the heads up 👍 I've rebased the PR and will mark it as Ready for Review.

File: .rubocop.yml [hunk 1]
```diff
@@ -55,7 +55,12 @@ Style/StringConcatenation:
 # for more discussion
 Style/FetchEnvVar:
   Enabled: false
-  
+
+# Rule enforced after migrating to Ruby 3, to prevent the new syntax which allows for hash values and keyword arguments
+# omission, such as `myMethod(x:, y:)` and `h = { a:, b: }`.
+Style/HashSyntax:
+  EnforcedShorthandSyntax: never
```
Comment (crazytonyli) [inline .rubocop.yml:62]: May I know why adding this rule? I think it'd good to adopt Ruby 3 features from now on?

**Review by crazytonyli — CHANGES_REQUESTED** [review]

File: .rubocop.yml (hunk 1 above)
Comment (mokagio) [reply .rubocop.yml:62]: @crazytonyli see paaHJt-5yi-p2

Comment (crazytonyli) [reply .rubocop.yml:62]: Thanks! I have seen that post, but it didn't come to me while reviewing the code. 🙈

**Review by crazytonyli — APPROVED** [review]
Thanks for the changes! Probably worth P2 about it, considering there may be linting issues from rubocop while we write new code. 😄

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -53,7 +53,6 @@ steps:
         matrix:
           setup:
             ruby:
-              - 2.7.4
```
Comment (crazytonyli) [inline .buildkite/pipeline.yml:56]: [The trunk branch protection rule](https://github.com/wordpress-mobile/release-toolkit/settings/branch_protection_rules/) needs to be changed to accommodate the pipeline change.

Comment (iangmaia) [reply .buildkite/pipeline.yml:56]: Yep, I've seen there's this required check that is pending, but was removed.

I don't seem to have access to the branch protection rules in `release-toolkit` though 🤔 . In fact, I don't see the Settings tab at all. Perhaps you or someone else can change it / change my role in the repo?

Comment (iangmaia) [reply .buildkite/pipeline.yml:56]: Done!

**Pull request merged**